slices/reduce.go

```diff
@@ -0,0 +1,25 @@
+package slices
+
+import "golang.org/x/exp/constraints"
+
+func ReduceWith[S ~[]E, E any, A any](s S, acc A, fn func(A, E) A) A {
```

Comment:
im skeptical of the `With`, it does not evoke anything to me, wdyt of `ReduceFrom` or `ReduceWithInitial`